akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -204,7 +223,33 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
           }
         }
       }
+
+      private def concatContinuationIntoHeaders(handler: SubStream, cont: ContinuationFrame) = {
+        val concatHeaderBlockFragment = handler.headers.headerBlockFragment ++ cont.payload
+        val moreCompleteHeadersFrame = HeadersFrame(handler.streamId, cont.endHeaders, cont.endHeaders, concatHeaderBlockFragment)
+        // update the SubStream we keep around
+        val moreCompleteHandler = handler.copy(headers = moreCompleteHeadersFrame)
+        incomingStreams += handler.streamId → moreCompleteHandler
+        moreCompleteHandler
+      }
+
+      def makeHttp2SubStream(handler: SubStream): Http2SubStream = {
+        val headers = handler.headers
+        val subStream = handler.outlet
+        if (headers.endStream && headers.endHeaders) {
+          Http2SubStream(headers, Source.empty)
+        } else {
+          // FIXME a bit naive but correct I think -- todo check the spec 
```

Comment:
well that's not enterely true since one *can* actually send a header/continuation at the end (after the last data frame). This actually replaces the chunked response, that send trailers at the end.